docs/api/build_proto.md

```diff
@@ -0,0 +1,346 @@
+
+    stable-x86_64-unknown-linux-gnu (default)
+    rustc 1.58.0 (02072b482 2022-01-11)
+    ```
+
+1.  Create project
+
+    ```bash
+    $ cargo new --lib vald-grpc
+    ```
+
+1.  Edit `Cargo.toml`
+
+    ```bash
+    $ cd vald-grpc
+    $ vim Cargo.toml
+    ---
+    [package]
+    name = "vald-grpc"
+    version = "0.1.0"
+    edition = "2021"
+
+    # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+    [[bin]]
+    name = "client"
+    path = "src/client.rs"
+
+    [dependencies]
+    tonic = "0.6.2"
+    tokio = "1.15"
+    prost = "0.9"
+    prost-types = "0.9"
+    hdf5 = "0.8.1"
+    chrono = "^0.4"
+
+    [build-dependencies]
+    tonic-build = "0.6.2"
+    ```
+
+1.  Download Vald proto files and dependence proto files
+
+    ```bash
+    // create proto file root dir
+    $ mkdir -p proto
+
+    // download vald proto files
+    $ git clone https://github.com/vdaas/vald \
+    && cp -R vald/apis/proto/v1/vald proto/vald \
+    && cp -R vald/apis/proto/v1/payload proto/payload \
+    && rm -rf vald
+
+    // download googleapis
+    $ git clone https://github.com/googleapis/googleapis \
+    && cp -R googleapis/google proto/google \
+    && rm -rf googleapis
+
+    // download protoc-gen-validate
+    $ git clone https://github.com/envoyproxy/protoc-gen-validate \
+    && mv protoc-gen-validate proto/protoc-gen-validate
+    ```
+
+1.  Fixing import path
+
+    ```bash
+    $ find proto/vald -type f -name "*.proto" | xargs sed -i "s/apis\/proto\/v1\///g"
+    $ find proto/vald -type f -name "*.proto" | xargs sed -i "s/github\.com\/googleapis\/googleapis\///g"
+    $ find proto/payload -type f -name "*.proto" | xargs sed -i "s/github\.com\/googleapis\/googleapis\///g"
+    $ find proto/payload -type f -name "*.proto" | xargs sed -i "s/github\.com\/envoyproxy\///g"
+    ```
+
+1.  Implement `build.rs` and Build proto
+
+    1. `build.rs`
+
+       ```rust
+       fn main() -> Result<(), Box<dyn std::error::Error>> {
+           let insert_proto = "./proto/vald/insert.proto";
+           let update_proto = "./proto/vald/update.proto";
+           let upsert_proto = "./proto/vald/upsert.proto";
+           let search_proto = "./proto/vald/search.proto";
+           let remove_proto = "./proto/vald/remove.proto";
+           let object_proto = "./proto/vald/object.proto";
+
+           tonic_build::configure()
+               .build_client(true)
+               .compile(
+                   &[
+                       insert_proto,
+                       update_proto,
+                       upsert_proto,
+                       search_proto,
+                       remove_proto,
+                       object_proto,
+                   ],
+                   &["./proto"],
+               )
+               .unwrap_or_else(|e| panic!("protobuf compile error: {}", e));
+           Ok(())
+       }
+       ```
+
+    1. build proto
+
+       ```bash
+       $ cargo build
```

Comment:
the following error occurred after I execute this command.
do you have any idea how to fix it?
```
> cargo build
    Updating crates.io index
   Compiling cfg-if v1.0.0
   Compiling proc-macro2 v1.0.36
   Compiling unicode-xid v0.2.2
   Compiling libc v0.2.116
......
   Compiling tower v0.4.11
   Compiling vald-grpc v0.1.0 (/Users/wdiu/go/src/github.com/vdaas/vald/vald-grpc)
   Compiling hyper v0.14.16
   Compiling hyper-timeout v0.4.1
   Compiling tonic v0.6.2
error: couldn't read src/client.rs: No such file or directory (os error 2)

error: could not compile `vald-grpc` due to previous error
```

Reply:
I can build it after I commented the `bin` section in `Cargo.toml`.

Reply:
it's true.
I have updated the document.